modules/c++/math.poly/include/math/poly/Fit.h

```diff
@@ -250,6 +251,79 @@ inline math::poly::TwoD<double> fit(size_t numRows,
 
     return fit(xm, ym, zm, nx, ny);
 }
+
+typedef math::linear::VectorN<3, double> Vector3;
+
+inline math::poly::OneD<Vector3> fit(
+    const math::linear::Vector<double>& xObs,
+    const math::linear::Vector<double>& yObs0,
+    const math::linear::Vector<double>& yObs1,
+    const math::linear::Vector<double>& yObs2,
+    size_t numCoeffs)
+{
+    size_t numObs = xObs.size();
+    if (yObs0.size() != numObs || yObs1.size() != numObs || yObs2.size() != numObs)
+    {
+        throw except::Exception(Ctxt("Must have the same number of observed y values as observed x values"));
+    }
+    if (numCoeffs < 1)
+    {
+        throw except::Exception(Ctxt("Number of coefficients must be >= 1"));
+    }
+
+    math::poly::OneD<double> fit0 = fit(xObs, yObs0, numCoeffs);
+    math::poly::OneD<double> fit1 = fit(xObs, yObs1, numCoeffs);
+    math::poly::OneD<double> fit2 = fit(xObs, yObs2, numCoeffs);
+
+    math::poly::OneD<Vector3> polyVector3 = math::poly::OneD<Vector3>(numCoeffs-1);
+    for (size_t i = 0; i< numCoeffs; i++)
+    {
+        Vector3 coeffs = Vector3();
```

Comment:
Change `i` to `ii` (avoid single character variable names)
Also, can't you say `Vector3& coeffs = polyVector3[i]` and then avoid the copy and last line?